Dockerfile

```diff
@@ -1,30 +1,38 @@
-# builder
-FROM golang:1.9.3-alpine3.7 AS builder
-
-WORKDIR /go/src/github.com/alicebob
-
+# Accept the Go version for the image to be set as a build argument.
+# Default to Go 1.12
+ARG GO_VERSION=1.12
 ARG ASPROM_VERSION=1.0.1
 
-RUN apk add --no-cache git \
-    && git clone https://github.com/alicebob/asprom.git \
-    && cd asprom \
-    && git fetch --all --tags --prune \
-    && git checkout tags/$ASPROM_VERSION
+# First stage: build the executable.
+FROM golang:${GO_VERSION}-alpine AS build
+
+# Create the user and group files that will be used in the running container to
+# run the process as an unprivileged user.
+RUN mkdir /user && \
+    echo '1000:x:65534:65534:1000:/:' > /user/passwd && \
+    echo '1000:x:65534:' > /user/group
 
-WORKDIR /go/src/github.com/alicebob/asprom
+# Install the Certificate-Authority certificates for the app to be able to make
+# calls to HTTPS endpoints.
+RUN apk add --no-cache ca-certificates git curl
 
-RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o asprom .
+WORKDIR /src
 
+COPY . .
 
+RUN go mod download && \
+    CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o asprom .
```

Comment:
Can you make this:

 RUN  CGO_ENABLED=0 GOOS=linux go build -mod=vendor -o asprom .

the `-a installsuffic cgo` is not needed anymore in recent Go-s. The `-mod=vendor` makes Go use the files from `./vendor/`, so you don't need to download anything anymore (the `go mod download`).

Reply:
fixed